numba/cpython/setobj.py

```diff
@@ -1188,10 +1227,12 @@ def set_constructor(context, builder, sig, args):
 
     # If the argument has a len(), preallocate the set so as to
     # avoid resizes.
+    # decref each item to avoid leak from iterator's incref
```

Comment:
I be more specific in this comment and say something like both `add` and `for_iter` incref each element, so we add a decref to avoid a leak.